projects/igniteui-angular/src/lib/services/transaction/igx-transaction.ts

```diff
@@ -47,7 +47,7 @@ export class IgxTransactionService<T extends Transaction, S extends State> exten
         if (!this._isPending) {
             this._undoStack.push([{ transaction, recordRef }]);
             this._redoStack = [];
-            this.onStateUpdate.emit();
+            this.onStateUpdate.emit({ origin: TransactionEventOrigin.ADD });
```

Comment:
what we push in `_unodStack` is actually an `action`. Add this action to the event arguments.